TraceInfo Function Panics If Tracing Is Not Enabled

Hi, I came across an issue today where TraceInfo() panics when called and EnableTracing has been called. (Version v2.1.0)

Example based on [SimpleGet](https://github.com/go-resty/resty#simple-get) (https://play.golang.org/p/n1EvoADpC3P)

```
	client := resty.New()

	resp, err := client.R().
		Get("https://httpbin.org/get")
...
	ti := resp.Request.TraceInfo()
	fmt.Println("DNSLookup    :", ti.DNSLookup)
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0xffffffff addr=0x0 pc=0x3b229a]

goroutine 1 [running]:
github.com/go-resty/resty/v2.(*Request).TraceInfo(0x896240, 0x80c110, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/tmp/gopath455282213/pkg/mod/github.com/go-resty/resty/v2@v2.1.0/request.go:533 +0x7a
main.main()
	/tmp/sandbox271512670/prog.go:28 +0x840
```

Running with EnableTrace wont panic (As expected)
```
	client := resty.New()

	resp, err := client.R().
		EnableTrace().
		Get("https://httpbin.org/get")
...
	ti := resp.Request.TraceInfo()
	fmt.Println("DNSLookup    :", ti.DNSLookup)
```
```
Request Trace Info:
DNSLookup    : 0s
```

This might be intentional behaviour but a simple nil check and maybe returning an error would be a cleaner solution and provide more control to the calling application. 

Thought I would ask and happy to add a fix if needed 🙂

Thanks

--- a/request.go
+++ b/request.go
@@ -525,10 +525,17 @@ func (r *Request) EnableTrace() *Request {
 }
 
 // TraceInfo method returns the trace info for the request.
+// If either the Client or Request EnableTrace function has not been called
+// prior to the request being made, an empty TraceInfo object will be returned.
 //
 // Since v2.0.0
 func (r *Request) TraceInfo() TraceInfo {
 	ct := r.clientTrace
+
+	if ct == nil {
+		return TraceInfo{}
+	}
+
 	return TraceInfo{
 		DNSLookup:     ct.dnsDone.Sub(ct.dnsStart),
 		ConnTime:      ct.gotConn.Sub(ct.getConn),

--- a/request_test.go
+++ b/request_test.go
@@ -1555,3 +1555,24 @@ func TestTraceInfo(t *testing.T) {
 	// for sake of hook funcs
 	_, _ = client.R().EnableTrace().Get("https://httpbin.org/get")
 }
+
+func TestTraceInfoWithoutEnableTrace(t *testing.T) {
+	ts := createGetServer(t)
+	defer ts.Close()
+
+	client := dc()
+	client.SetHostURL(ts.URL)
+	for _, u := range []string{"/", "/json", "/long-text", "/long-json"} {
+		resp, err := client.R().Get(u)
+		assertNil(t, err)
+		assertNotNil(t, resp)
+
+		tr := resp.Request.TraceInfo()
+		assertEqual(t, true, tr.DNSLookup == 0)
+		assertEqual(t, true, tr.ConnTime == 0)
+		assertEqual(t, true, tr.TLSHandshake == 0)
+		assertEqual(t, true, tr.ServerTime == 0)
+		assertEqual(t, true, tr.ResponseTime == 0)
+		assertEqual(t, true, tr.TotalTime == 0)
+	}
+}

